shared/video-sdk/authentication-workflow/project-implementation/cpp.mdx

```diff
@@ -1,34 +1,51 @@
+1. **Enable the user to specify a channel**
+
+    Add a text box to the user interface:
+
+    1. Go to menu **View** > **Other Windows** > **Resource View**.
+        
+        If the **Resource View** window isn't the top-most window, select the **Resource View** tab to bring it to the top.
+
+    1. In **Resource View**, go to **AgoraImplementation.rc** > **Dialog** and double-click `IDD_AGORAIMPLEMENTATION_DIALOG`.
+
+        The resource opens inside the **Dialog Editor**. 
+    
+    1. Click **View** > **ToolBox**. The **Toolbox** opens.
+
+    1. From **ToolBox**, drag **Edit Control** to the surface of the **Dialog Editor**.
+
 1. **Import the required class**
 
-    To use the `CHttpFile` class in your project, add the following to the list of `#include` statements in `CAgoraImplementationDlg.cpp`:
+    To use the `CHttpFile` class in your project, add the following to the list of `#include` statements in `AgoraImplementationDlg.cpp`:
 
     ```cpp
-    #include afxinet.h
+    #include "afxinet.h"
     ```
 2. **Add variables for your connection to the token server**
 
     Declare the variables you need to specify the user id, token-server URL and the token expire time. In `AgoraImplementationDlg.h`, add the following declarations to the `CAgoraImplementationDlg` class after `channelName` declaration:
 
     ``` cpp
     int tokenRole = 1; // The token role: Broadcaster or Audience
-	std::string serverUrl = <The base URL to your token server>; //For example, https://agora-token-service-production-92ff.up.railway.app"
+	std::string serverUrl = "<The base URL to your token server>"; //For example, https://agora-token-service-production-92ff.up.railway.app"
 	int expireTime = 60; //Expire time in Seconds.
 	uid_t uid = 1; // An integer that identifies the user
     ```
     Make sure you specify the token server URL in exactly the same format as shown in the example.
 
 4. **Retrieve a token from the server**
 
-    Use a GET request to retrieve an authentication token for a specific channel from the token server. In `AgoraImplementationDlg.cpp`, add the following `fetchToken()` method before `onJoinChannelSuccess()`:
-
+    Use a GET request to retrieve an authentication token for a specific channel from the token server. To implement a GET request using `CHttpFile`, in `AgoraImplementationDlg.cpp`, add the following before `setupVideoSDKEngine`:
```

Comment:
I defined this function globally, not part of the class. That's why kept it at the start. Although this is definition I can keep it any place, just need to put before calling the fetchToken()